integration_test/third_party_apps_data/applications/dcgm/centos_rhel/install

```diff
@@ -1,54 +1,84 @@
+    sudo yum install -y kernel-devel-$(uname -r) pciutils gcc make wget yum-utils 
+    local DRIVER_VERSION=535.129.03
+    echo "Installing NVIDIA Data Center driver $DRIVER_VERSION"
+    curl -fSsl -O https://us.download.nvidia.com/tesla/$DRIVER_VERSION/NVIDIA-Linux-x86_64-$DRIVER_VERSION.run
+    sudo bash ./NVIDIA-Linux-x86_64-$DRIVER_VERSION.run --silent
+    verify_driver
+}
 
-# Install DCGM
-VERSION_ID=${VERSION_ID%%.*}
-sudo yum-config-manager \
-    --add-repo http://developer.download.nvidia.com/compute/cuda/repos/rhel$VERSION_ID/x86_64/cuda-rhel$VERSION_ID.repo
-sudo yum clean expire-cache
-sudo yum install -y datacenter-gpu-manager
-sudo systemctl --now enable nvidia-dcgm
-
-# check DCGM service running and load profiling module
-dcgmi discovery --list
+setup_repo() {
+    sudo yum install -y yum-utils
+    sudo yum-config-manager \
+        --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel$VERSION_ID/x86_64/cuda-rhel$VERSION_ID.repo
+    sudo yum clean all
+}
+
+install_dcgm() {
+    # Ref: https://docs.nvidia.com/datacenter/dcgm/latest/user-guide/getting-started.html#rhel-centos-rocky-linux
+    setup_repo
+    sudo yum install -y datacenter-gpu-manager
+    sudo systemctl --now enable nvidia-dcgm 
+
+    # check DCGM service running and load profiling module
+    dcgmi discovery --list
+}
+
+try_install() {
+    for install_method in "$@"; do
+        echo "Installing NVIDIA driver with $install_method..."
+        # Temporarily disable the "-e" option so that failures in the function
+        # won't cause this script to exit; also run the installation method in a
+        # subshell with -e so that the function itself will exit immediately if 
+        # any command encounters error
+        set +e ; ( set -e; $install_method ) ; exit_code=$?
+        set -e
+        if [ $exit_code -eq 0 ]; then
+            echo "NVIDIA driver has been installed successfully with $install_method."
+            return 
+        fi
+    done
+    echo "NVIDIA driver cannot be installed; all installation methods failed."
+    return 1
+}
+
+
+handle_rhel7() {
+    install_driver_from_package_manager() {
+        # Ref: https://developer.nvidia.com/cuda-12-2-2-download-archive?target_os=Linux&target_arch=x86_64&Distribution=RHEL&target_version=7&target_type=rpm_network
+        setup_repo
+        sudo yum -y install nvidia-driver-latest-dkms
+        verify_driver
+    }
+
+    try_install install_driver_from_runfile install_driver_from_package_manager
+}
+
+handle_common() {
+    install_driver_from_package_manager() {
+        # Ref: https://developer.nvidia.com/cuda-12-2-2-download-archive?target_os=Linux&target_arch=x86_64&Distribution=RHEL&target_version=8&target_type=rpm_network
+        setup_repo
+        sudo yum -y module install nvidia-driver:latest
+        verify_driver
+    }
+
+    try_install install_driver_from_package_manager install_driver_from_runfile
+}
```

Comment:
You can make it even shorter, e.g.:
```suggestion
install_driver_from_package_manager() {
    setup_repo
    install_driver_package
    verify_driver
}

handle_rhel7() {
    install_driver_package() {
        # Ref: https://developer.nvidia.com/cuda-12-2-2-download-archive?target_os=Linux&target_arch=x86_64&Distribution=RHEL&target_version=7&target_type=rpm_network
        sudo yum -y install nvidia-driver-latest-dkms
    }

    try_install install_driver_from_runfile install_driver_from_package_manager
}

handle_common() {
    install_driver_package() {
        # Ref: https://developer.nvidia.com/cuda-12-2-2-download-archive?target_os=Linux&target_arch=x86_64&Distribution=RHEL&target_version=8&target_type=rpm_network
        sudo yum -y module install nvidia-driver:latest
    }

    try_install install_driver_from_package_manager install_driver_from_runfile
}
```
Also, do we ever expect the driver _verification_ to fail after it's been installed (as opposed to the installation itself)? If not, you can pull out `verify_driver` to after the `try_install` step.

Reply:
Done! Thanks!

`verify_driver` can fail even when the installation returns no error. When this happens, we want `try_install` to try the next method (runfile). Added comments to `verify_driver` function with potential error messages. 

The runfile method would error out if it find an existing package manager driver installation, so in the latest commit I have modified the runfile method to try to remove yum installation first.